internal/authz/sub_repo_perms.go

```diff
@@ -319,6 +313,39 @@ func SubRepoEnabled(checker SubRepoPermissionChecker) bool {
 	return checker != nil && checker.Enabled()
 }
 
+// CanReadAllPaths returns true if the actor can read all paths.
+func CanReadAllPaths(ctx context.Context, checker SubRepoPermissionChecker, repo api.RepoName, paths []string) (bool, error) {
+	if !SubRepoEnabled(checker) {
+		return true, nil
+	}
+	a := actor.FromContext(ctx)
+	if a.IsInternal() {
+		return true, nil
+	}
+	if !a.IsAuthenticated() {
+		return false, &ErrUnauthenticated{}
+	}
+
+	var perms Perms
+	var err error
+	c := RepoContent{
+		Repo: repo,
+	}
+
+	for _, p := range paths {
+		c.Path = p
+		perms, err = checker.Permissions(ctx, a.UID, c)
```

Comment:
I don't think there is any need to pre-declare perms and err here

```suggestion
	c := RepoContent{
		Repo: repo,
	}

	for _, p := range paths {
		c.Path = p
		perms, err := checker.Permissions(ctx, a.UID, c)
```